sktime/transformations/base.py

```diff
@@ -48,93 +104,537 @@ class BaseTransformer(BaseEstimator):
+            | `Panel`  | `Primitives` | `pd.DataFrame`         |
+            | `Series` | `Series`     | `Series`               |
+            | `Panel`  | `Series`     | `Panel`                |
+            | `Series` | `Panel`      | `Panel`                |
+        instances in return correspond to instances in `X`
+        combinations not in the table are currently not supported
         """
+        X = _handle_alias(X, Z)
         # Non-optimized default implementation; override when a better
         # method is possible for a given algorithm.
-        if X is None:
-            # Fit method of arity 1 (unsupervised transformation)
-            return self.fit(Z).transform(Z)
-        else:
-            # Fit method of arity 2 (supervised transformation)
-            return self.fit(Z, X).transform(Z)
+        return self.fit(X, y).transform(X, y)
 
     # def inverse_transform(self, Z, X=None):
     #     raise NotImplementedError("abstract method")
     #
     # def update(self, Z, X=None, update_params=False):
     #     raise NotImplementedError("abstract method")
 
+    def _fit(self, X, y=None):
+        """
+        Fit transformer to X and y.
+
+        core logic
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _fit must support all types in it
+            Data to fit transform to
+        y : Series or Panel of mtype y_inner_mtype, optional, default=None
+            Additional data, e.g., labels for tarnsformation
+
+        Returns
+        -------
+        self: a fitted instance of the estimator
+        """
+        # default fit is "no fitting happens"
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        core logic
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _transform must support all types in it
+            Data to be transformed
+        y : Series or Panel, optional (default=None)
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        transformed version of X
+        type depends on type of X and scitype:transform-output tag:
+            |          | `transform`  |                        |
+            |   `X`    |  `-output`   |     type of return     |
+            |----------|--------------|------------------------|
+            | `Series` | `Primitives` | `pd.DataFrame` (1-row) |
+            | `Panel`  | `Primitives` | `pd.DataFrame`         |
+            | `Series` | `Series`     | `Series`               |
+            | `Panel`  | `Series`     | `Panel`                |
+            | `Series` | `Panel`      | `Panel`                |
+        instances in return correspond to instances in `X`
+        combinations not in the table are currently not supported
```

Comment:
ah, yes, I should have been more precise - of the output is `Primitives`, the return is always `pd.DataFrame`. Otherwise, if the input is `Panel`, the output is `Panel`.

Not a problem, since we have both types of transformers in one.